src/loan/LendingTerm.sol

```diff
@@ -0,0 +1,367 @@
+        return _totalBorrowsStored + interestPerYear * (block.timestamp - totalBorrowsLastUpdate) / YEAR;
+    }
+
+    /// @notice call fee of a loan, in CREDIT base units
+    function getLoanCallFee(bytes32 loanId) public view returns (uint256) {
+        Loan storage loan = loans[loanId];
+        uint256 originationTime = loan.originationTime;
+
+        if (originationTime == 0) {
+            return 0;
+        }
+
+        if (loan.closeTime != 0) {
+            return 0;
+        }
+
+        return loan.borrowAmount * callFee / 1e18;
+    }
+
+    /// @notice outstanding borrowed amount of a loan, including interests
+    function getLoanDebt(bytes32 loanId) public view returns (uint256) {
+        Loan storage loan = loans[loanId];
+        uint256 originationTime = loan.originationTime;
+
+        if (originationTime == 0) {
+            return 0;
+        }
+
+        if (loan.closeTime != 0) {
+            return 0;
+        }
+
+        // compute interest owed
+        uint256 borrowAmount = loan.borrowAmount;
+        uint256 interest = borrowAmount * interestRate * (block.timestamp - originationTime) / YEAR / 1e18;
+        uint256 loanDebt = borrowAmount + interest;
+
+        return loanDebt;
+    }
+
+    /// @notice initiate a new loan
+    function borrow(
+        uint256 borrowAmount,
+        uint256 collateralAmount
+    ) external whenNotPaused returns (bytes32 loanId) {
+        require(borrowAmount != 0, "LendingTerm: cannot borrow 0");
+        require(collateralAmount != 0, "LendingTerm: cannot stake 0");
+
+        loanId = keccak256(abi.encode(msg.sender, address(this), block.timestamp));
+
+        // check that the loan doesn't already exist
+        require(loans[loanId].originationTime == 0, "LendingTerm: loan exists");
+
+        // check that enough collateral is provided
+        uint256 maxBorrow = collateralAmount * creditPerCollateralToken / 1e18;
+        require(borrowAmount <= maxBorrow, "LendingTerm: not enough collateral");
+
+        // check that ltvBuffer is respected
+        uint256 maxBorrowLtv = maxBorrow * 1e18 / (1e18 + ltvBuffer);
+        require(borrowAmount <= maxBorrowLtv, "LendingTerm: not enough LTV buffer");
+
+        // check that this lending term is active
+        address _guildToken = guildToken;
+        require(GuildToken(_guildToken).isGauge(address(this)), "LendingTerm: terms unavailable");
+
+        // check the debt ceiling & hardcap
+        uint256 _totalBorrows = totalBorrows();
+        require(_totalBorrows + borrowAmount <= hardCap, "LendingTerm: hardcap reached");
+        uint256 _totalSupply = ERC20(creditToken).totalSupply();
+        if (_totalSupply != 0) {
+            uint256 debtCeiling = GuildToken(_guildToken).calculateGaugeAllocation(address(this), _totalSupply + borrowAmount);
+            require(_totalBorrows + borrowAmount <= debtCeiling, "LendingTerm: debt ceiling reached");
+        }

```

Comment:
what happens if the system is perfectly balanced ? say 50%/50% votes on 2 terms and both terms have the same borrow amount